$OpenBSD: patch-src_datautils_cpp,v 1.1 2015/05/21 05:52:10 dcoppa Exp $

Be careful about the lifetime of stack variables!
ptr is pointing to free'd memory, causing an infinite loop when
parsing the video duration property
(upstream git commit 9b1ff7b5b800f5e34f39bf9289d0bd0ca89949c4)

http://blog.aaronballman.com/2011/07/returning-stack-based-values/

https://wiki.qt.io/Qt_project_org_faq#How_can_I_convert_a_QString_to_char.2A_and_vice_versa.3F

--- src/datautils.cpp.orig	Tue May 12 11:27:07 2015
+++ src/datautils.cpp	Tue May 12 11:29:04 2015
@@ -50,7 +50,9 @@ uint DataUtils::parseIsoPeriod(const QString &isoPerio
 
     uint days = 0, hours = 0, minutes = 0, seconds = 0;
 
-    const char *ptr = isoPeriod.toStdString().c_str();
+    QByteArray ba = isoPeriod.toLocal8Bit();
+    const char *ptr = ba.data();
+
     while (*ptr) {
         if(*ptr == 'P' || *ptr == 'T') {
             ptr++;
